### PR TITLE
Optimized quick plotting

### PR DESCRIPTION
**Pull request by studywolf**
- also added plt.show(block=False) command to have it work on Ubuntu

instead of redrawing the whole image only update those parts that have changed, based on code from http://bastibe.de/2013-05-30-speeding-up-matplotlib.html

### CODE REVIEW

**Comment by tcstewar** [comment]
Cool.  If this was done as an alternate display mode (i.e. you have to set `display_mode='ubuntu_quick'` for this to do anything, otherwise it'd fall back to the original version) then I'd be fine to merge it now.  As it is, though, I'd need to test it on a bunch of different systems to see how that functions on them.  which I'll hopefully get around to after the summer school.

**Comment by studywolf** [comment]
done!

**Comment by studywolf** [comment]
bump!

File: nstbot/retinabot.py [hunk 1]
```diff
@@ -115,12 +122,21 @@ def image_loop(self, decay, display_mode):
                 scatter.set_color(c)
 
             if display_mode == 'quick':
-                # this is much faster, but doesn't work on all systems
-                fig.canvas.draw()
+                # this is faster, but doesn't work on all systems
+                fig.canvas_draw()
```
Comment (tcstewar) [inline nstbot/retinabot.py]: this should be `fig.canvas.draw()`   :)

**Comment by tcstewar** [comment]
with the one typo I noted above, this runs fine for me on windows.  Has it been tested on OS X?

**Comment by tcstewar** [comment]
> Has it been tested on OS X?

(note: I'm not too worried about making sure it's been tested on OS X: I'm pretty sure I never tested the code on it originally.  I'm just curious if it runs...  :)  )

**Comment by studywolf** [comment]
that's a very good question! i avoid macs like the plague so I'm not sure! 

change made! and within a month this time!

**Comment by studywolf** [comment]
bump!

**Comment by studywolf** [comment]
bump!

**Comment by tcstewar** [comment]
> bump!

Oops..  yes, I should really do this.  I just need to find a camera to test it on....  it looks like there's one pointing at the robot arm that I probably shouldn't disturb -- do you know where the other one is?

**Comment by studywolf** [comment]
you can grab either of those ones! it's fine :)

**Comment by tcstewar** [comment]
okay, I've confirmed this still works under windows, and I assume you're happy with how it works in Ubuntu, so I'll go ahead and merge...  :)

**Comment by studywolf** [comment]
success!!

**Pull request merged**